### ISSUE TITLE
It seem I need a C or C++ compiler

### ISSUE THREAD

**Issue by jisraelpo** [andlabs/libui#253]
So suggest me what can I do please:

The C compiler identification is unknown
The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:30 (project):
  The CMAKE_C_COMPILER:

    cl

  is not a full path and was not found in the PATH.

  To use the NMake generator with Visual C++, cmake must be run from a shell
  that can use the compiler cl from the command line.  This environment is
  unable to invoke the cl compiler.  To fix this problem, run cmake from the
  Visual Studio Command Prompt (vcvarsall.bat).

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.

CMake Error at CMakeLists.txt:30 (project):
  The CMAKE_CXX_COMPILER:

    cl

  is not a full path and was not found in the PATH.

  To use the NMake generator with Visual C++, cmake must be run from a shell
  that can use the compiler cl from the command line.  This environment is
  unable to invoke the cl compiler.  To fix this problem, run cmake from the
  Visual Studio Command Prompt (vcvarsall.bat).

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.

Configuring incomplete, errors occurred!
See also "C:/libui-master/build/CMakeFiles/CMakeOutput.log".
See also "C:/libui-master/build/CMakeFiles/CMakeError.log".

**Comment by andlabs** [comment]
Yes, this is a C-based project and thus requires a C compiler to build. Are you looking for something in particular?

**Comment by jisraelpo** [comment]
Yes, I want to test this framework in Julia Lang and then in go lang


So what need todo ?


Thanks for you reply.


________________________________
De: Pietro Gagliardi <notifications@github.com>
Enviado: martes, 7 de febrero de 2017 12:14 a. m.
Para: andlabs/libui
Cc: jisraelpo; Author
Asunto: Re: [andlabs/libui] It seem I need a C or C++ compiler (#253)


Yes, this is a C-based project and thus requires a C compiler to build. Are you looking for something in particular?

**Comment by andlabs** [comment]
If the Julia bindings don't ship with a precompiled version of the library, then yes you will need a C/C++ compiler.

My Go bindings do ship with precompiled binaries, but you still need the C compiler for cgo to work.

**Comment by joa-quim** [comment]
Right, I didn't include a pre-compiled dll (use Windows only) but can get one if needed.
Also note that this a work in progress and am waiting for new developments in libui, namely and very needed for me, the ability to plot images. BTW @andlabs any news of when will this be possible?

**Comment by jisraelpo** [comment]
Hi again.


Personally I think Julia will be a very important languge in a short future, under its slogan "As fast as C, as easy as Python" it could beat several strong today languages.


I am just playing with Julia and in my research I saw libui is the best library to develop desktop app right now. I think too desktop apps will never disappear (Maybe because I am a Delphi Jedi) So I would like to continue playing with Julia.


Then can you drive me in the way I can use this lib ?


PD. I undestand this lib is as young as Julia thus can not be deployed in a production stage.


________________________________
De: Joaquim <notifications@github.com>
Enviado: martes, 7 de febrero de 2017 02:24 p. m.
Para: andlabs/libui
Cc: jisraelpo; Author
Asunto: Re: [andlabs/libui] It seems I need a C or C++ compiler (#253)


Right, I didn't include a pre-compiled dll (use Windows only) but can get one if needed.
Also note that this a work in progress and am waiting for new developments in libui, namely and very needed for me, the ability to plot images. BTW @andlabs<https://github.com/andlabs> any news of when will this be possible?

**Comment by joa-quim** [comment]
I added a link at the libui.jl github site to a 64 bits libui.dll. Regarding the help on how to use it in Julia, I'm afraid the help level is about the same as for using libui itself. You have an example ``controlgallery.jl`` in the examples directory that you can use to learn. I ported it from the C example here.

**Comment by jisraelpo** [comment]
Thanks again Joaquim for your fast reply


Then...


1.- I need to download Libui.dll and put in the same julia\bin directory of my computer.


2.- Install the package in julia: Pkg.clone("Libui")


3.- Use it: using Libui




________________________________
De: Joaquim <notifications@github.com>
Enviado: martes, 7 de febrero de 2017 05:29 p. m.
Para: andlabs/libui
Cc: jisraelpo; Author
Asunto: Re: [andlabs/libui] It seems I need a C or C++ compiler (#253)


I added a link at the libui.jl github site to a 64 bits libui.dll. Regarding the help on how to use it in Julia, I'm afraid the help level is about the same as for using libui itself. You have an example controlgallery.jl in the examples directory that you can use to learn. I ported it from the C example here.

**Comment by joa-quim** [comment]
Basically, yes. Though I wouldn't store libui.dll in julia\bin. Let it be in it's own place but make sure that place is in your path so Julia can find it.

**Comment by andlabs** [comment]
Any updates on this front? I've switched the build system to meson, if that'll make integration any easier.
